clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1188,6 +1190,20 @@ private void close(long timeout, TimeUnit timeUnit, boolean swallowException) {
         }
     }
 
+    private static Map<String, Object> propsToMap(Properties properties) {
+        Map<String, Object> map = new HashMap<>(properties.size());
+        for (Map.Entry<Object, Object> entry : properties.entrySet()) {
+            if (entry.getKey() instanceof String) {
+                String k = (String) entry.getKey();
+                map.put(k, properties.get(k));
+            }
```

Comment:
It seems that the checkstyle failed but all unit tests have passed. I can modify the code slightly to fix the checkstyle failure before merging it.